AuditRulesMonitor.cpp

```diff
@@ -0,0 +1,228 @@
+//
+// Created by tad on 3/18/19.
+//
+
+#include "AuditRulesMonitor.h"
+
+#include "Logger.h"
+#include "ExecUtil.h"
+
+void AuditRulesMonitor::run() {
+    Logger::Info("AuditRulesMonitor: Starting");
+    while(!_sleep(1000)) {
+        auto now = std::chrono::steady_clock::now();
+
+        if (now - _last_auoms_file_check > std::chrono::seconds(60)) {
+            _last_auoms_file_check = now;
+            get_desired_rules();
+        }
+
+        if (now - _last_audit_file_check > std::chrono::seconds(5*60)) {
+            _last_audit_file_check = now;
+            check_file_rules();
+        }
+
+        if (!check_kernel_rules()) {
```

Comment:
Do we want / need this check every second? Given the rules file is only checked every minute.